webapp/src/containers/Rhs/projectDetails/index.tsx

```diff
@@ -71,14 +79,23 @@ const ProjectDetails = (projectDetails: ProjectDetails) => {
 
     const project: FetchSubscriptionList = {project: projectDetails.projectName};
     const {data, isLoading} = usePlugin.getApiState(plugin_constants.pluginApiServiceConfigs.getSubscriptionList.apiServiceName, project);
-    const subscriptionList = data as SubscriptionDetails[];
 
     // Fetch subscription list
     const fetchSubscriptionList = () => usePlugin.makeApiRequest(
         plugin_constants.pluginApiServiceConfigs.getSubscriptionList.apiServiceName,
         project,
     );
 
+    // Handles switch toggle.
+    const handleToggle = () => {
+        if (showAllSubscriptions) {
+            setSubscriptionList(getCurrentChannelSubscriptions(data as SubscriptionDetails[], currentChannelId));
+        } else {
+            setSubscriptionList(data as SubscriptionDetails[]);
```

Comment:
We should've set the type for the `data` to `SubscriptionDetails[]` in the starting only rather than typing it again and again.